subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/MavenToolchainsInstallationSupplierTest.groovy

```diff
@@ -171,6 +171,128 @@ class MavenToolchainsInstallationSupplierTest extends Specification {
+                                <toolchain>
+                                <type>jdk</type>
+                                <configuration>
+                                <jdkHome>${env.JDK16}</jdkHome>
+                                </configuration>
+                                </toolchain>
+                                <toolchain>
+                                <type>jdk</type>
+                                <configuration>
+                                <jdkHome>${env.JDK17}</jdkHome>
+                                </configuration>
+                                </toolchain>
+                                </toolchains>'''
+        def toolchains = mavenHome.createFile(new File("toolchains.xml"))
+        toolchains.write(toolchain)
+        def supplier = useProperty ? createSupplier(toolchains.getCanonicalPath(), null) : createSupplier(null, userHome)
+
+        when:
+        def directories = supplier.get()
+
+        then:
+        directoriesAsStablePaths(directories) == stablePaths([
+            new File("/usr/lib/jvm/adoptopenjdk-16.jdk").absolutePath,
+            new File("/usr/lib/jvm/temurin-17.jdk").absolutePath
+        ])
+        directories*.source == ["Maven Toolchains", "Maven Toolchains"]
+
+        where:
+        useProperty << [true, false]
+    }
+
+    def "supplies multiple paths for multiple toolchains with un/resolvable environment variable"(boolean useProperty) {
+        given:
+        def toolchain = '''<toolchains>
+                                <toolchain>
+                                <type>jdk</type>
+                                <configuration>
+                                <jdkHome>${env.JDK16}</jdkHome>
+                                </configuration>
+                                </toolchain>
+                                <toolchain>
+                                <type>jdk</type>
+                                <configuration>
+                                <jdkHome>${env.JAVA_UNKNOWN}</jdkHome>
+                                </configuration>
+                                </toolchain>
+                                </toolchains>'''
+        def toolchains = mavenHome.createFile(new File("toolchains.xml"))
+        toolchains.write(toolchain)
+        def supplier = useProperty ? createSupplier(toolchains.getCanonicalPath(), null) : createSupplier(null, userHome)
+
+        when:
+        def directories = supplier.get()
+        
+
+        then:
+        directoriesAsStablePaths(directories) == stablePaths([
+            new File("/usr/lib/jvm/adoptopenjdk-16.jdk").absolutePath,
+            new File("\${env.JAVA_UNKNOWN}").absolutePath
+        ]).sort()
+        directories*.source == ["Maven Toolchains", "Maven Toolchains"]
+
+        where:
+        useProperty << [true, false]
+    }
+
+    def "supplies no installations for toolchains file with non-resolvable environment variable"(boolean useProperty) {
```

Comment:
:x: The test title needs to change since the behavior changed.

Reply:
Thanks for pointing this out ! changed it.